packages/d/duckdb/xmake.lua

```diff
@@ -3,38 +3,110 @@ package("duckdb")
     set_description("DuckDB is an in-process SQL OLAP Database Management System")
     set_license("MIT")
 
-    add_urls("https://github.com/duckdb/duckdb/releases/download/$(version)/libduckdb-src.zip",
-             "https://github.com/duckdb/duckdb.git")
-    add_versions("v1.2.1", "c7f21c12039e951dbb74e064ba218a4ca8b4b0a612d40c62d95a858ecaf2fb53")
-    add_versions("v1.1.3", "66beec9f299c56f508fec3d647faa86a596d39b2d4e26bd6991b29d07b818f83")
-    add_versions("v1.1.2", "8b30ec65addfe423fc18f8403bd958953d01a3193ec8b3bc2e43073433d94e47")
-    add_versions("v1.1.1", "48f1ca566ae0f73fb536aec22d599917f47c26ea7ae300380e7ae3f39e29af4d")
-    add_versions("v1.0.0", "482c7f090cac4408eed5b02708b6a54168c1875c2c6d8042d8344edee3f70eb7")
-    add_versions("v0.10.3", "f22ed5058188c81886dfdda80c6c46b7df2ed0a623a348c78ac8372451c84497")
-    add_versions("v0.10.2", "6bab203dc2e10432edbefda7be7043f73f17c8898ba81ce3aa2319e7e2d5af10")
-    add_versions("v0.10.1", "70ec6ffefd9a04bf9fcdc1a4949611f9633f10f0e3b9cead1425b926a54d0f89")
-    add_versions("v0.10.0", "385e27aa67712813e4a07389465c4c5c45c431d97cddd35713b8a306d2a86f2d")
-
-    on_install("macosx", "linux", function (package)
-        io.writefile("xmake.lua", [[
-            add_rules("mode.debug", "mode.release")
-            set_languages("c++17")
-            target("duckdb")
-                set_kind("$(kind)")
-                add_files("duckdb.cpp")
-                add_headerfiles("duckdb.hpp", "duckdb.h")
-        ]])
-        import("package.tools.xmake").install(package)
+    on_source("windows", "macosx", "linux", function(package)
+        local precompiled = false
+        if package:is_plat("windows") then
+            if package:is_arch("x64", "x86_64") then
+                precompiled = true
+                package:add("urls", "https://github.com/duckdb/duckdb/releases/download/$(version)/libduckdb-windows-amd64.zip")
+                package:add("versions", "v1.2.1", "c977230a701d6b274fe63166868c8e2bc4ec23cf42c44f08f22bdcd8ed4a8e31")
+            elseif package:is_arch("arm64") then
+                precompiled = true
+                package:add("urls", "https://github.com/duckdb/duckdb/releases/download/$(version)/libduckdb-windows-arm64.zip")
+                package:add("versions", "v1.2.1", "de02a9e3e64b41ea7844aa04ad7e72519ca131327e739cd344a6ba476fb31c0f")
+            end
+        elseif package:is_plat("linux") then
+            if package:config("shared") then
```

Comment:
we cannot use package:config in on_source. all configs are not inited here.

Reply:
So how can I differentiate between precompiled packages in static/shared, in the on_source?

Reply:
no way now.